### ISSUE TITLE
Cannot find module '@redwoodjs/testing/web' when running Storybook

### ISSUE THREAD

**Issue by Philzen** [redwoodjs/redwood#5054]
Having followed the tutorial up to chapter 5, i am getting this error in the dev console when i run `yarn rw storybook`:

![grafik](https://user-images.githubusercontent.com/1634615/161774185-c3a9a392-f9a8-4721-82d3-328661000b68.png)

All of the stories just render a blank page in the UI, the Cells and Pages are not even listed.

I am unsure whether this is typescript related or not – from my experience any typescript errors are gracefully handled when compiling to JS, and this seems to be a webpack error message. 

However, when i run the js-based https://github.com/redwoodjs/redwood-tutorial there is no such error – the dependencies listed in `package.json` are identical to my project, which you can check out here: https://github.com/Philzen/redwoodblog-tutorial-typescript

Any ideas what's going on here greatly appreciated.

**Comment by Philzen** [comment]
@jtoar Thanks heaps. I hope i'm not holding you off too much from other things with this, but maybe it's worthwhile understanding the root cause of this.

One noteworthy thing is that in v1.0.0 the Cells seem to generate imports in `*.test.tsx` than in earlier versions – at least when comparing it to the redwood tutorial the imports there are just `@redwoodjs/testing`, while in the ones i just created two days ago it's importing `@redwoodjs/testing/web`.

However, that shouldn't matter as there is a mapping for that in `web/tsconfig.json`. Also tried deleting both tsconfigs and running `yarn rw setup tsconfig`, but they come out exactly the same again.

**Comment by jtoar** [comment]
@Philzen any time, and not at all—we just released so feedback is super valuable. This one seems legit; I can reproduce it pretty easily on a new project.

Re the difference in the imports: great catch, I totally missed that. It's curious that those are different. I'll share this with the team and bump it in priority.

**Comment by Philzen** [comment]
Thanks @jtoar, so i can be "glad" in the sense that this may actually be helpful.

I ran into another bump when i synced my repo further with the redwood tutorial repo. I thought it was due to me changing the imports back to `@redwoodjs/testing` (without the web) in 16161a83ff2a7c4a129a06f2d7b07101f90cf632, but it actually happens when i sync the stories in 28f4697ba7dd41f3e96654ef307119eecd2c2aa8. 

The offending line is: 
```js
  mockCurrentUser({ id: 42, email: 'rob@redwoodjs.com' })
```

As soon as this line appears in my story, storybook won't even startup (or just crash on recompilation if already running) with:

```terminal
webpack building...
70% sealing finish module graph/home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:165
		return moduleGraph.getParentModule(this).buildMeta.strictHarmonyModule
		                                        ^

TypeError: Cannot read properties of undefined (reading 'buildMeta')
    at HarmonyImportSpecifierDependency._getEffectiveExportPresenceLevel (/home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:165:43)
    at HarmonyImportSpecifierDependency.getWarnings (/home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:176:32)
    at Compilation.reportDependencyErrorsAndWarnings (/home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/Compilation.js:3132:24)
    at /home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/Compilation.js:2729:28
    at eval (eval at create (/home/phil/prog/testing/rwb-tutorial/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:50:1)
    at /home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:385:11
    at /home/phil/prog/testing/rwb-tutorial/node_modules/neo-async/async.js:2830:7
    at Object.each (/home/phil/prog/testing/rwb-tutorial/node_modules/neo-async/async.js:2850:39)
    at /home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:361:18
    at /home/phil/prog/testing/rwb-tutorial/node_modules/neo-async/async.js:2830:7
    at Object.each (/home/phil/prog/testing/rwb-tutorial/node_modules/neo-async/async.js:2850:39)
    at /home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/FlagDependencyExportsPlugin.js:51:16
    at Hook.eval [as callAsync] (eval at create (/home/phil/prog/testing/rwb-tutorial/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:42:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/phil/prog/testing/rwb-tutorial/node_modules/tapable/lib/Hook.js:18:14)
    at Compilation.finish (/home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/Compilation.js:2714:28)
    at /home/phil/prog/testing/rwb-tutorial/node_modules/@storybook/builder-webpack5/node_modules/webpack/lib/Compiler.js:1182:19
```

Not sure if this is related, but i have a hunch it might.

**Comment by ebuckley** [comment]
After trying quite a few things to fix this, I have come up with the following workaround.

1.  modify the `web/jsconfig.json` to match the https://github.com/redwoodjs/redwood-tutorial/blob/main/web/jsconfig.json project.
2. `rm -r ./redwood`
3. `yarn rw storybook`

I'm not familiar enough with the project to know if all three are neccessary, But I am able to happily continue my tutorial experience now that I have done the above modifications.

**Comment by thedavidprice** [comment]
Thanks all! 

@virtuoushub any chance you have time to take a look at this as well?

**Comment by thedavidprice** [comment]
> One noteworthy thing is that in v1.0.0 the Cells seem to generate different imports in *.test.tsx than in earlier versions – at least when comparing it to redwoodjs/redwood-tutorial the imports there are just @redwoodjs/testing, while in the cell tests i just generated two days ago the import is @redwoodjs/testing/web.

^^ @dac09 any chance this has something to do with babel config?

**Comment by thedavidprice** [comment]
Core Team, I'm not sure if it's possible, but I'd really like us to try and "catch" this bug in our CI — maybe demonstrate via a PR?

Also of Note: possibly two problems happening here. @Philzen is reporting the issue on TS project (correct?) and it looks like @ebuckley is on JS (correct?). Definitely something suspicious the imports changing and the resolution via removing `./redwood` (aka generated types).

## To Do
- [ ] Redwood-Tutorial Repo: update api/ and web/ jsconfig.json to reflect current templates

**Comment by realStandal** [comment]
Running into the Storybook error after upgrading my application from `1.0.0-rc.6` to `1.0.0`.

What's peculiar is that @Philzen's solution didn't work and @ebuckley's doesn't seem like what I ended up doing to solve it.

My solution didn't need any changes to the codebase, it ended up being (what I presume) was a dependency hiccup. I don't want to post mine and add to the confusion (I honestly am not 100% confident in _how_ I solved it), but if someone else comes across this under similar circumstances/it ends up being needed, happy to share.

**Comment by dac09** [comment]
> > One noteworthy thing is that in v1.0.0 the Cells seem to generate different imports in *.test.tsx than in earlier versions – at least when comparing it to redwoodjs/redwood-tutorial the imports there are just @redwoodjs/testing, while in the cell tests i just generated two days ago the import is @redwoodjs/testing/web.
> ^^ @dac09 any chance this has something to do with babel config?

I don't think this is related to babel or the import - I made this change probably 5 or 8 releases ago - when we released the webhook api function testing. All we do is alias redwoodjs/testing to redwoodjs/testing/web on the web side for backwards compatibility reasons. Our tutorials should probably use the more accurate `@redwoodjs/testing/web` import path though!


I think we have two different issues though, from the same problem:
1. Adding mockCurrentUser to stories results in the error
```
TypeError: Cannot read properties of undefined (reading 'buildMeta')
```
This error happens before storybook even launches. 

2. `@redwoodjs/testing/web` isn't being resolved. These are indeed because of the globally available "devTimeAutoImports" loaded [in the webpack config](https://github.com/redwoodjs/redwood/blob/main/packages/core/config/webpack.common.js#L133)

This error happens once storybook launches, but you go to a story.

Why? Still don't know...

**Comment by dac09** [comment]
OK FOUND ISSUE! This solves both issues.

Tried a bunch of theories, but pulling on the thread from this issue, it looks like this happens when storybook for some reason picks webpack 4 accidentally (extrapolating from [this thread](https://github.com/storybookjs/storybook/issues/15252)) - When there are multiple versions of webpack 5 (in our case, redwood is using 5.7**0**.0, but storybook wants 5.7**1**.0) - see this output from yarn why
```
gitpod /workspace/redwoodblog-tutorial-typescript (main) $ yarn why webpack
├─ @redwoodjs/core@npm:1.0.0
│  └─ webpack@npm:5.70.0 [b5b1e] (via npm:5.70.0 [b5b1e]) 👈
│
├─ @storybook/builder-webpack4@npm:6.4.19
│  └─ webpack@npm:4.46.0 (via npm:4)
.
.

│  └─ webpack@npm:5.71.0 (via npm:^5.9.0)  👈
│
├─ @storybook/manager-webpack5@npm:6.4.19 [7f8c1]
│  └─ webpack@npm:5.71.0 [0dc39] (via npm:^5 [0dc39]) 👈
.
.
```

The solution for now, is to force the resolution of webpack 5, but we should issue a patch with the updated webpack version so theres no mismatch. **Note** that its not the updated webpack that solves the problem, its definitely a storybook thing, because even if you set the webpack version to 5.70.0 it still works fine

In your root package.json
```
  "resolutions": {
    "webpack": "5.71.0"
  },
```

**Comment by virtuoushub** [comment]
@dac09 great find!

> Thanks all!
> 
> @virtuoushub any chance you have time to take a look at this as well?

In case it helps others; I briefly took a look, and it seemed to be webpack related. I was looking at `@storybook/builder-webpack5/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:165`, but was unable to really dig into the problem and start to make sense of what I was seeing. Really appreciate @dac09's work on this, as well as @Philzen's helpful info along the way.

**Comment by thedavidprice** [comment]
@dac09 Very interesting and great find.

Supposedly Storybook 6.5 will mitigate this type of issue ongoing... Was supposed to be out last month :(

**Comment by thedavidprice** [comment]
Will be included in v1.0.1
